src/Queue.ts

```diff
@@ -182,6 +184,20 @@ export class Queue {
     stop() {
         this.isActive = false;
     }
+
+    /**
+     * cancel running job
+     */
+    cancelJob(jobId: string, exception?: Error) {
+      const promise = this.runningJobPromises[jobId]
+      if (promise && typeof promise[CANCEL] === 'function') {
+        promise[CANCEL](exception || new Error(`canceled`))
+      } else if(!promise[CANCEL]){
+        console.warn("Worker does not have a cancel method implemented")
```

Comment:
I think the job should be removed from jobstore if cancel is not implement or not running